pkgs/by-name/le/lexical/package.nix

```
@@ -0,0 +1,56 @@
+{ lib
+, beamPackages
+, fetchFromGitHub
+, writeScript
+, elixir
+}:
+
+beamPackages.mixRelease rec {
+  pname = "lexical";
+  version = "0.5.2";
+
+  src = fetchFromGitHub {
+    owner = "lexical-lsp";
+    repo = "lexical";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-HWqwJ7PAz80bm6YeDG84hLWPE11n06K98GOyeDQWZWU=";
+  };
+
+  mixFodDeps = beamPackages.fetchMixDeps {
+    inherit pname version src;
+
+    sha256 = "sha256-G0mT+rvXZWLJIMfrhxq3TXt26wDImayu44wGEYJ+3CE=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    mix do compile --no-deps-check, package --path "$out"
+
+    runHook postInstall
+  '';
+
+  preFixup = let
+    activate_version_manager = writeScript "activate_version_manager.sh" ''
+      true
+    '';
+  in ''
+    substituteInPlace "$out/bin/start_lexical.sh" --replace 'elixir_command=' 'elixir_command="${elixir}/bin/"'
+    rm "$out/bin/activate_version_manager.sh"
+    ln -s ${activate_version_manager} "$out/bin/activate_version_manager.sh"
+
+    mv "$out/bin" "$out/binsh"
+
+    makeWrapper "$out/binsh/start_lexical.sh" "$out/bin/lexical" --set RELEASE_COOKIE lexical
```

```suggestion
  postInstall = ''
    substituteInPlace "$out/bin/start_lexical.sh" --replace 'elixir_command=' 'elixir_command="${elixir}/bin/"'

    mv "$out/bin" "$out/libexec"

    makeWrapper "$out/libexec/start_lexical.sh" "$out/bin/lexical" --set RELEASE_COOKIE lexical
```

Upstream could maybe clean this up some?
https://github.com/lexical-lsp/lexical/blob/bb1c1e3f66160653f95bd05af7589297233fec96/bin/start_lexical.sh#L7

`libexec` is a bit more standard than `binsh`, so better to use this. I considered getting rid of the extra directory completely, but I like only exposing `bin/lexical` to users.

---

I made the change.